src/main/java/com/redhat/sast/api/v1/resource/DashboardWebSocketResource.java

```diff
@@ -0,0 +1,100 @@
+package com.redhat.sast.api.v1.resource;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.websocket.OnClose;
+import jakarta.websocket.OnError;
+import jakarta.websocket.OnMessage;
+import jakarta.websocket.OnOpen;
+import jakarta.websocket.Session;
+import jakarta.websocket.server.ServerEndpoint;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * WebSocket endpoint for real-time dashboard updates.
+ *
+ * Broadcasts events when:
+ * - Job status changes
+ * - Batch progress updates
+ * - OSH scans collected/failed
+ */
+@ServerEndpoint("/ws/dashboard")
+@ApplicationScoped
+@Slf4j
+public class DashboardWebSocketResource {
+
+    private static final Map<String, Session> sessions = new ConcurrentHashMap<>();
+    private final ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+
+    @OnOpen
+    public void onOpen(Session session) {
+        sessions.put(session.getId(), session);
+        LOGGER.debug("Dashboard WebSocket connected: {} (total: {})", session.getId(), sessions.size());
```

Comment:
I'd not put any log statement for event methods. It's useful for local debugging, but not inside a deployed application. Please remove similar statements in other methods in this class.